src/Avalonia.Controls/Notifications/Notification.cs

```diff
@@ -0,0 +1,176 @@
+                            PseudoClasses.Add(":success");
+                            break;
+
+                        case NotificationType.Warning:
+                            PseudoClasses.Add(":warning");
+                            break;
+                    }
+                });
+        }
+
+        /// <summary>
+        /// Determines if the notification is already closing.
+        /// </summary>
+        public bool IsClosing
+        {
+            get { return _isClosing; }
+            private set { SetAndRaise(IsClosingProperty, ref _isClosing, value); }
+        }
+
+        public static readonly DirectProperty<Notification, bool> IsClosingProperty =
+            AvaloniaProperty.RegisterDirect<Notification, bool>(nameof(IsClosing), o => o.IsClosing);
+
+        /// <summary>
+        /// Determines if the notification is closed.
+        /// </summary>
+        public bool IsClosed
+        {
+            get { return _isClosed; }
+            set { SetAndRaise(IsClosedProperty, ref _isClosed, value); }
+        }
+
+        /// <summary>
+        /// Defines the <see cref="IsClosed"/> property.
+        /// </summary>
+        public static readonly DirectProperty<Notification, bool> IsClosedProperty =
+            AvaloniaProperty.RegisterDirect<Notification, bool>(nameof(IsClosed), o => o.IsClosed, (o, v) => o.IsClosed = v);
+
+        /// <summary>
+        /// Defines the <see cref="NotificationCloseInvoked"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> NotificationCloseInvokedEvent =
+            RoutedEvent.Register<Notification, RoutedEventArgs>(nameof(NotificationCloseInvoked), RoutingStrategies.Bubble);
+
+        /// <summary>
+        /// Defines the <see cref="NotificationClosed"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> NotificationClosedEvent =
+            RoutedEvent.Register<Notification, RoutedEventArgs>(nameof(NotificationClosed), RoutingStrategies.Bubble);
+
+        /// <summary>
+        /// Raised when notification close event is invoked.
+        /// </summary>
+        public event EventHandler<RoutedEventArgs> NotificationCloseInvoked
+        {
+            add { AddHandler(NotificationCloseInvokedEvent, value); }
+            remove { RemoveHandler(NotificationCloseInvokedEvent, value); }
+        }
+
+        public event EventHandler<RoutedEventArgs> NotificationClosed
+        {
+            add { AddHandler(NotificationClosedEvent, value); }
+            remove { RemoveHandler(NotificationClosedEvent, value); }
+        }
+
+        public static bool GetCloseOnClick(Notification obj)
+        {
+            return (bool)obj.GetValue(CloseOnClickProperty);
+        }
+
+        public static void SetCloseOnClick(Notification obj, bool value)
+        {
+            obj.SetValue(CloseOnClickProperty, value);
+        }
+
+        public static readonly AvaloniaProperty CloseOnClickProperty =
```

Comment:
Why is this property an attached property?

Also the host control is declared as `Button` but the static getter/setter takes a `Notification`.